src/ContractInteract/GatewayLib.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+const Web3 = require('web3');
+
+const AbiBinProvider = require('../AbiBinProvider');
+const Utils = require('../utils/Utils');
+
+const ContractName = 'GatewayLib';
+
+/**
+ * Contract interact for GatewayLib.
+ *
+ * As GatewayLib is a library contract, this only serves for deployment.
+ */
+class GatewayLib {
+  /**
+   * Constructor for GatewayLib.
+   *
+   * @param {Object} web3 Web3 object.
+   * @param {string} libraryAddress GatewayLib contract address.
+   */
+  constructor(web3, libraryAddress) {
+    if (web3 instanceof Web3) {
+      this.web3 = web3;
+    } else {
+      const err = new TypeError(
+        "Mandatory Parameter 'web3' is missing or invalid",
+      );
+      throw err;
+    }
+
+    if (!Web3.utils.isAddress(libraryAddress)) {
+      const err = new TypeError(
+        "Mandatory Parameter 'libraryAddress' is missing or invalid.",
+      );
+      throw err;
+    }
+
+    this.address = libraryAddress;
+  }
+
+  /**
+   * Deploys a GatewayLib contract.
+   *
+   * @param {Web3} web3 Web3 object.
+   * @param {string} merklePatriciaProof Address of MerklePatriciaProof contract
+   *                 to link into the contract bytecode.
+   * @param {Object} txOptions Transaction options.
+   *
+   * @returns {Promise<GatewayLib>} Promise containing the GatewayLib
+   *                                         instance that has been deployed.
+   */
+  static async deploy(web3, merklePatriciaProof, txOptions) {
```

Comment:
Maybe
```suggestion
  static async deploy(web3, merklePatriciaProofContract, txOptions) {
```
or
```suggestion
  static async deploy(web3, merklePatriciaProofAddress, txOptions) {
```
?

I was confused at first thinking "why does it need a Merkle Patricia proof...?"
Same for raw transaction?

Also applies to other classes...

Reply:
I'll make sure that the `*Address` variant is used consistently for the library contracts.